addons/event/controllers/main.py

```diff
@@ -10,10 +10,10 @@ class EventController(Controller):
 
     @route(['''/event/<model("event.event"):event>/ics'''], type='http', auth="public")
     def event_ics_file(self, event, **kwargs):
+        lang = request.env.user.lang
         if request.env.user._is_public():
-            frontend_lang = request.httprequest.cookies.get('frontend_lang')
-            if frontend_lang:
-                event = event.with_context(lang=frontend_lang)
+            lang = request.httprequest.cookies.get('frontend_lang')
```

Comment:
:wave: Hello

I think we can just use `request.lang` now? It handles both cases and it does a bunch of checks to make sure we actually get a lang in the end